src/Sentry/ISentryJsonSerializable.cs

```diff
@@ -19,12 +20,18 @@ public interface ISentryJsonSerializable
 
 internal static class JsonSerializableExtensions
 {
-    public static void WriteToFile(this ISentryJsonSerializable serializable, string filePath, IDiagnosticLogger? logger)
+    public static void WriteToFile(this ISentryJsonSerializable serializable, IFileSystem fileSystem, string filePath, IDiagnosticLogger? logger)
     {
-        using var file = File.Create(filePath);
+        var result = fileSystem.CreateFileForWriting(filePath, out var file);
+        if (result is not FileOperationResult.Success)
+        {
+            return;
+        }
+
         using var writer = new Utf8JsonWriter(file);
 
         serializable.WriteTo(writer, logger);
         writer.Flush();
+        file.Dispose();
```

Comment:
this couldn’t have `using` anymore? I’m concerned an exception will keep dangling filesa